lib/scope.js

```diff
@@ -120,7 +120,73 @@ module.exports = {
     expect( elem.constructor.name, `Did not find anything where the ${ ordinal } "${ label_text }" ${ roles } should have been. ${ elem.constructor.name }` ).to.equal('ElementHandle');
 
     return elem;
-  },  // Ends getSpecifiedChoice
+  },  // Ends getSpecifiedChoice()
+
+  tapElement: async function tapElement(scope, elem) {
+    /* Tap a given element, detect navigation, and do appropriate related things. */
+    expect( elem, 'That element cannot be found, so it cannot be tapped.' ).to.exist;
+
+    let start_url = await scope.page.url()
+
+    let winner;
+    if (elem) {
+      winner = await Promise.race([
+        Promise.all([  // Error loads page, so no need to detect
+          // Click with no navigation will end immediately
+          elem[ scope.activate ](),
+          scope.page.waitForNavigation({waitUntil: 'domcontentloaded'}),
+        ]),
+        scope.page.waitForSelector('.alert-danger'),
+        scope.page.waitForSelector('.da-has-error'),
+      ]);
+    }
+
+    // Not sure how to detect navigation landing at same page.
+    let end_url = await scope.page.url();
+    let navigated = start_url !== end_url;
+    // Might be able to handle this in `scope.afterStep`
+    if ( navigated ) {  // if (winner.constructor.name === 'ElementHandle') {
+      // If navigation, wait for 200ms to let things settle down
+      await scope.afterStep(scope, {waitForTimeout: 200, navigated: navigated});
+    } else {
+      // If stay on page, wait for `show if` or feedback elements to be move around
+      await scope.afterStep(scope, {waitForShowIf: true, navigated: navigated});
+    }
+  },  // Ends tapElement()
+
+  checkboxNameAttribute: async function ( scope, var_name, choice_name ) {
+    /* Covert a checkbox variable and choice value to its base64
+    *    attribute names. In da, a (non-yesno/noyes) checkbox `name`
+    *    attribute is made of a combination of its variable's name
+    *    and it's choice value's name, plus a little more.
+    * 
+    * @param {object} scope - The global namespace object
+    * @param {string} var_name - Variable name of the variable being set
+    * @param {string} choice_name - Value that will be set to true for the variable
+    * 
+    * Examples:
+    * checkboxNameAttribute( scope, 'eviction_all_reasons', 'nonpayment' );
+    * // "ZXZpY3Rpb25fYWxsX3JlYXNvbnNbQidibTl1Y0dGNWJXVnVkQSdd"
```

Comment:
fix the return value - it should be an object of 2 strings

Reply:
This function is no longer being used anywhere and should be removed! Yay!